pandora-server-directory/src/database/mongoDb.ts

```diff
@@ -518,9 +519,58 @@ export default class MongoDatabase implements PandoraDatabase {
 
 	private async _doMigrations(): Promise<void> {
 		// insert migration code here
+		for await (const account of this._accounts.find().stream()) {
+			if (account.directMessages == null)
+				continue;
+
+			const directMessages: typeof account.directMessages = [];
+			for (const dm of account.directMessages) {
+				const migrated = { ...dm };
+				if ('account' in migrated)
+					delete migrated.account;
+				if ('displayName' in migrated)
+					delete migrated.displayName;
+
+				directMessages.push(migrated);
+			}
+
+			Assert(directMessages.length === account.directMessages.length);
+			const { matchedCount, modifiedCount } = await this._accounts.updateOne({ id: account.id }, { $set: { directMessages } });
+			Assert(matchedCount === 1 && modifiedCount === 1);
+		}
+
+		if (await this._accountContacts.countDocuments() > 0) {
+			logger.warning('Account contacts already migrated');
+			return;
+		}
+
+		const relationships = this._db.collection<DatabaseAccountRelationshipOld>('relationships');
+		const relationshipsCount = await relationships.countDocuments();
+
+		for await (const rel of relationships.find().stream()) {
+			const contact = {
```

Comment:
Nit: Please add an explicit type to make sure it matches exactly~ (I would have loved to see running it through a Zod schema the most, but this specific thing is only a type, not a schema right now).